app/controllers/devise_token_auth/registrations_controller.rb

```diff
@@ -105,6 +95,18 @@ def account_update_params
 
     protected
 
+    def build_resource
```

Comment:
Seeing the `@resource` instance variable over and over again, makes me wonder if this business logic for building a resource belongs inside the model (or model concern) instead?

For example:

~~~ruby
# In the resource_class model (via ActiveSupport::Concern, or otherwise):
module ClassMethods
  def new_registration(params)
    params[:email].try(:downcase!) if case_insensitive_keys.include?(:email)
    new(params.except(:confirm_success_url))
  end
end
~~~

~~~ruby
# In DeviseTokenAuth::RegistrationsController#create:
# ...

def create
  @resource = resource_class.new_registration(sign_up_params)

  # ...
end

# ...
~~~

You could still override the extracted method to customize the behavior, just inside your model instead of the controller.

Reply:
On second thought, given how I refactored things in my example above, I think a bulk of the existing logic involves manipulating parameters, and most of it could live inside the `#sign_up_params` protected method that handles the strong_parameters.  That would leave just a plain `@resource = resource_class.new(sign_up_params)` call in the `#create` action.

... Of course, that would not leave a method for you to override anymore.

Another approach might be to extract that `@resource = resource_class.new(sign_up_params)` into a `before_action` callback.  Then you could maintain the ability to customize the behavior by overriding the callback method.

Reply:
- about the concern: yes my first idea was to have something like the [ResourceFinder](https://github.com/lynndylanhurley/devise_token_auth/blob/master/app/controllers/devise_token_auth/concerns/resource_finder.rb#L20)Concern. But for building. And i'd prefer have it stay in the Controllers rather than the models.